ui/StatusQ/src/StatusQ/Popups/Dialog/StatusDialog.qml

```diff
@@ -101,4 +113,9 @@ Dialog {
             }
         }
     }
+
+    onClosed: {
+        if (root.destroyOnClose)
+            root.destroy();
+    }
```

Comment:
It does not belong here, it is not the responsibility of the Dialog.  The lifetime of objects should be managed by the entity that creates them.